tests/integration/ambient/waypoint_test.go

```diff
@@ -105,16 +100,9 @@ func TestWaypoint(t *testing.T) {
 					nsConfig.Name(),
 					"--service-account",
 					sa,
+					"--wait-ready",
```

Comment:
@hanxiaop I have added the --wait-ready for this integration test you wrote.
We need to keep in mind though that this will be slightly less performant than the old readiness check since now we are doing ["install waypoint s1", "check ready for waypoint s1", "install waypoint s2", "check ready for waypoint s2" ...]
instead of ["install waypoint s1", "install waypoint s2", "check ready for waypoint s1", "check ready for waypoint s2" ...].
Given that we're only deploying three waypoints in this test, which are relatively quick to reach a ready state, the impact of using the --wait-ready flag would be minimal. However, I believe it's still worth mentioning this aspect :)

Reply:
Makes sense, thanks for the explaination.